services/static-webserver/client/source/class/osparc/dashboard/StudyBrowser.js

```diff
@@ -170,31 +171,35 @@ qx.Class.define("osparc.dashboard.StudyBrowser", {
     },
 
     __reloadFolders: function() {
-      if (!osparc.auth.Manager.getInstance().isLoggedIn()) {
+      if (
+        !osparc.auth.Manager.getInstance().isLoggedIn() ||
+        !osparc.utils.DisabledPlugins.isFoldersEnabled() ||
+        this.__loadingFolders) {
         return;
       }
-
-      if (osparc.utils.DisabledPlugins.isFoldersEnabled()) {
-        const folderId = this.getCurrentFolderId();
-        const workspaceId = this.getCurrentWorkspaceId();
-        if (workspaceId === -1 || workspaceId === -2) {
-          return;
-        }
-        this.__setFoldersToList([]);
-        osparc.store.Folders.getInstance().fetchFolders(folderId, workspaceId, this.getOrderBy())
-          .then(folders => {
-            this.__setFoldersToList(folders);
-          })
-          .catch(console.error);
+      const workspaceId = this.getCurrentWorkspaceId();
+      if (workspaceId === -1 || workspaceId === -2) {
+        return;
       }
+
+      this.__loadingFolders = true;
+      this.__setFoldersToList([]);
+      const folderId = this.getCurrentFolderId();
+      osparc.store.Folders.getInstance().fetchFolders(folderId, workspaceId, this.getOrderBy())
+        .then(folders => {
+          this.__setFoldersToList(folders);
+        })
+        .catch(console.error)
+        .finally(() => this.__loadingFolders = null);
     },
 
     __reloadStudies: function() {
-      if (this._loadingResourcesBtn.isFetching() || !osparc.auth.Manager.getInstance().isLoggedIn()) {
-        return;
-      }
       const workspaceId = this.getCurrentWorkspaceId();
-      if (workspaceId === -1) { // shared workspace listing
+      if (
+        !osparc.auth.Manager.getInstance().isLoggedIn() ||
+        workspaceId === -1 || // listing workspaces
```

Comment:
is this a special sentinel for workspaces? Is this used as well in the web API query or it is only front-end?

Reply:
frontend only, it's the current way to define the context of the StudyBrowser, it will change in a coming PR